src/plugins/auto/src/plugin.cpp

```diff
@@ -575,91 +575,73 @@ std::list<DeviceInformation> Plugin::get_valid_device(
     if (meta_devices.empty()) {
         OPENVINO_THROW("No available device to select in ", get_device_name());
     }
-
+    bool is_default_list = true;
+    std::list<DeviceInformation> valid_devices;
     std::list<DeviceInformation> CPU;
     std::list<DeviceInformation> dGPU;
     std::list<DeviceInformation> iGPU;
-    std::list<DeviceInformation> MYRIAD;
-    std::list<DeviceInformation> VPU;
+    std::list<DeviceInformation> Others;
+    auto is_supported_model = [this, model_precision](const std::string& device_name) {
+        // Check if candidate device supported the specified model precision.
+        auto capability = get_core()->get_property(device_name, ov::device::capabilities);
+        auto support_model = std::find(capability.begin(), capability.end(), (model_precision));
+        bool is_support_fp16 =
+            model_precision == "FP32" && std::find(capability.begin(), capability.end(), ("FP16")) != capability.end();
+        return support_model != capability.end() || is_support_fp16;
+    };
 
-    for (auto& item : meta_devices) {
-        if (item.device_name.find("CPU") == 0) {
-            CPU.push_back(item);
-            continue;
-        }
-        if (item.device_name.find("MYRIAD") == 0) {
-            MYRIAD.push_back(item);
-            continue;
-        }
-        if (item.device_name.find("VPU") == 0) {
-            VPU.push_back(item);
+    for (auto&& device_info : meta_devices) {
+        if (device_info.device_priority > 0)
+            is_default_list = false;
+        // check if device support this model precision
+        if (!is_supported_model(device_info.device_name) && meta_devices.size() > 1)
```

Comment:
```suggestion
        // Add device to valid_devices if it is the only device in meta_devices, no matter it supports required precision or not.
        if (!is_supported_model(device_info.device_name) && meta_devices.size() > 1)
```
Do I understand correctly?